employee/admin.py

```diff
@@ -1,16 +1,162 @@
 
+# Remove fields from fieldset
 
-class EmployeeAdmin(admin.ModelAdmin):
-    exclude = ("createdBy", "updatedBy")
-    list_display = ("picture_tag", "fullname", "join_date",  "designation")
-    list_filter = ("designation", "join_date")
+
+def remove_from_fieldsets(fieldsets, fields):
+    for fieldset in fieldsets:
+        for field in fields:
+            if field in fieldset[1]['fields']:
+                new_fields = []
+                for new_field in fieldset[1]['fields']:
+                    if not new_field in fields:
+                        new_fields.append(new_field)
+
+                fieldset[1]['fields'] = tuple(new_fields)
+                break
+
+# Remove fieldsets
+
+
+def remove_fieldsets(fieldsets, fieldsetname):
+    new_fieldsets = []
+    for fieldset in fieldsets:
+        if fieldset[0] is not fieldsetname:
+            new_fieldsets.append(fieldset)
+    return new_fieldsets
+
+
+class EmployeeAdmin(UserAdmin):
+    add_form = EmployeeCreationForm
+    form = EmployeeChangeForm
+    model = Employee
+    fieldsets = (
+        (None, {'fields': ('fullname', 'email', 'password')}),
+        ('Permissions', {'fields': ('is_staff',
+                                    'is_superuser', 'is_active')}),
+        ('Personal Information',
+            {
+                'fields':
+                (
+                    'dob',
+                    'gender',
+                    'phone_number',
+                    'address',
+                    'pan_no',
+                    'citizenship_no',
+                    'designation',
+                    'picture',
+                    'pan_no_document',
+                    'citizenship_document',
+                    'department',
+                )
+            }
+         ),
+    )
+    add_fieldsets = (
+        (None, {
+            'classes': ('wide',),
+            'fields': (
+                'fullname',
+                'email',
+                'password1',
+                'password2',
+                'dob',
+                'gender',
+                'phone_number',
+                'address',
+                'pan_no',
+                'citizenship_no',
+                'designation',
+                'picture',
+                'pan_no_document',
+                'citizenship_document',
+                'department',
+                'date_joined',
+            )
+        }
+        ),
+    )
+    exclude = ("created_by", "updated_by")
+    list_display = (
+        "picture_tag",
+        "fullname",
+        "date_joined",
+        "designation",
+        'department',
+        'is_active'
+    )
+    list_filter = ("designation", "department", "date_joined")
     search_fields = ["fullname"]
+    ordering = ('email',)
+
+    def get_queryset(self, request):
+        if request.user.is_staff is True and request.user.is_superuser is False:
+            return Employee.objects.filter(email=request.user.email)
+        return Employee.objects.all()
```

Comment:
This will make more sense.
```suggestion
        if request.user.is_superuser:
            return Employee.objects.all()
        return Employee.objects.filter(email=request.user.email)
```

Also, `request.user.is_staff` is not required since users can't log in to the admin panel without it.
https://docs.djangoproject.com/en/3.1/ref/contrib/admin/
![2021-01-21-093047](https://user-images.githubusercontent.com/7059255/105277354-5d7e9e80-5bcb-11eb-918a-3f98d39c5755.png)

Reply:
Worked according to the suggestion.